src/sql/magic.py

```diff
@@ -499,6 +503,8 @@ def interactive_execute_wrapper(**kwargs):
         if args.creator:
             args.creator = user_ns[args.creator]
 
+        validate_nonidentifier_connection(args, command)
```

Comment:
I can't remember exactly where's the logic that parses this:

```
import duckdb
some_conn = duckdb.connect()

%sql some_conn
```

(looks up `some_conn` in the user's namespace and use it as a connection)

I think it's in the command.py file. it'll be better to move this logic to that function because so all the logic is in one place

Reply:
[fixed](https://github.com/ploomber/jupysql/pull/821/files#diff-769097f5e36be57ddec10c78df0900db8c1e254f5485bc94367aa7fcb07991c5R72) : Moved to `SQLCommand` in `command.py` where it determines a connection